### PR TITLE
Add GraphQL

### PR DESCRIPTION
**Pull request by nathaliebru**

### CODE REVIEW

**Comment by sabinewst** [comment]
localhost:3333 isn't running when running `npm run dev`, change your `json-server` script to `json-server --watch db.json --port 3004 & json-graphql-server db.json --p 3333` :)

File: src/GraphQL/Mutations.tsx [hunk 1]
```diff
@@ -0,0 +1,31 @@
+import { gql } from "@apollo/client";
```
Comment (sabinewst) [inline src/GraphQL/Mutations.tsx]: I think you could also make this a `.gql` file instead of `.tsx`

File: src/components/team-members-component/TeamMembersComponent.tsx [hunk 2]
```diff
@@ -58,7 +58,7 @@ export const TeamMembersComponent = (props: TeamMembersComponentProps) => {
           </select>
         </label>
 
-        {teamMembers
+        {[...teamMembers]
```
Comment (sabinewst) [inline src/components/team-members-component/TeamMembersComponent.tsx:63]: Why this change?

Comment (nathaliebru) [reply src/components/team-members-component/TeamMembersComponent.tsx:63]: Need this against mutation.

File: pages/index.tsx [hunk 3]
```diff
@@ -1,38 +1,43 @@
...
 
-import { getTimeEntries } from "../src/services/get-time-entries";
-import { NotFoundError } from "../src/error/not-found-error";
-
 import * as Types from "../src/types/TimeEntry.types";
-import { getClients } from "../src/services/get-clients";
 
 interface HomepageProps {
   timeEntries: Types.TimeEntry[];
   clients: [];
```
Comment (saravdberg) [inline pages/index.tsx]: Alphabetic order

File: pages/team-members/index.tsx [hunk 4]
```diff
@@ -1,10 +1,11 @@
 import React from "react";
+import { gql } from "@apollo/client";
+import client from "../../apollo-client";
```
Comment (brambrink) [inline pages/team-members/index.tsx:1]: Use named export here

File: src/components/new-time-entry/NewTimeEntry.tsx [hunk 5]
```diff
@@ -1,10 +1,10 @@
 import React, { useContext, useRef, useState } from "react";
+import { useMutation } from "@apollo/client";
+import { ADD_TIME_ENTRY } from "../../GraphQL/Mutations";
```
Comment (brambrink) [inline src/components/new-time-entry/NewTimeEntry.tsx]: I would put this line with the Styles and the Types as the upper two imports are external

File: src/components/time-entries/TimeEntries.tsx [hunk 6]
```diff
@@ -1,5 +1,6 @@
 /* eslint-disable react/destructuring-assignment */
 import React, { useContext, useEffect, useState } from "react";
```
Comment (brambrink) [inline src/components/time-entries/TimeEntries.tsx]: You don't need to import React

File: pages/team-members/index.tsx [hunk 7]
```diff
@@ -13,19 +14,24 @@ interface teamMemberProps {
...
-  const teamMembersResponse = await getTeamMembers();
-
-  if (teamMembersResponse instanceof NotFoundError) {
-    return {
-      props: {
-        teamMembers: [],
-      },
-    };
-  }
+  const { data } = await client.query({
```
Comment (brambrink) [inline pages/team-members/index.tsx:16]: Consider adding this to your GraphQL folder

File: src/GraphQL/Mutations.tsx (hunk 1 above)
Comment (brambrink) [inline src/GraphQL/Mutations.tsx]: I would name the folder `graphql` or maybe `graph-ql` to be consistent with the other folders

**Review by brambrink — COMMENTED** [review]
Looks very good 🤩 Only minor comments 💬

File: pages/team-members/index.tsx (hunk 4 above)
Comment (nathaliebru) [reply pages/team-members/index.tsx:1]: What do you mean with named export?

**Review by saravdberg — APPROVED** [review]

File: pages/team-members/index.tsx (hunk 4 above)
Comment (brambrink) [reply pages/team-members/index.tsx:1]: Instead of default exporting client from apollo-client.js, you could only export it. Then you import is using { client } here. It is not very important though I think!

**Review by brambrink — APPROVED** [review]
Nice 🏆
